src/kakarot/account.cairo

```diff
@@ -439,7 +439,17 @@ namespace Account {
             tempvar range_check_ptr = range_check_ptr;
         }
 
-        tempvar is_not_done = is_le(next_i + 1, bytecode_len);
+        // Checks if i+1 > bytecode_len. In this case, ap is filled with 0.
+        tempvar a = bytecode_len - (next_i + 1);
+        %{ memory[ap] = 0 if 0 <= (ids.a % PRIME) < range_check_builtin.bound else 1 %}
+        ap += 1;
+        // is_not_done == 1 <=> i+1 <= bytecode_len.
+        if ([ap - 1] == 0) {
+            tempvar is_not_done = 1;
+        } else {
+            tempvar is_not_done = 0;
+        }
+
```

Comment:
isn't it just `tempvar is_not_done = 1 - [ap - 1]` ?

Reply:
and maybe ?

```
        tempvar a = (next_i + 1) - bytecode_len;
        %{ memory[ap] = 0 if 0 <= (ids.a % PRIME) < range_check_builtin.bound else 1 %}
        ap += 1;
        let is_not_done = [ap - 1];
```

Reply:
> and maybe ?
> 
> ```
>         tempvar a = (next_i + 1) - bytecode_len;
>         %{ memory[ap] = 0 if 0 <= (ids.a % PRIME) < range_check_builtin.bound else 1 %}
>         ap += 1;
>         let is_not_done = [ap - 1];
> ```

The issue is that jmp if [... ] is always of the form jmp if x!=0, we cannot do jump if x==0, and unfortunately the whitelisted hint is in reverse...

Reply:
I am not sure to follow. I don't really care about the value, the decision is actually binary (0 or not 0).
I have written `a` to be your `-a` to have first something like

```
tempvar a = (next_i + 1) - bytecode_len;
%{ memory[ap] = 0 if 0 <= (ids.a % PRIME) < range_check_builtin.bound else 1 %}
ap += 1;
// is_not_done == 1 <=> i+1 <= bytecode_len.
if ([ap - 1] == 0) {
    tempvar is_not_done = 0;
} else {
    tempvar is_not_done = 1;
}
```

but eventually `is_not_done` is just used as you said, to compare to `0`, so this could as well be

```
tempvar is_not_done = [ap - 1];
```

and using `let` because it's not necessary to add a step here

Reply:
```suggestion
        // Check termination with whitelisted hint
        // ap = 0 <=> i >= bytecode_len.
        tempvar a = next_i - bytecode_len;
        %{ memory[ap] = 0 if 0 <= (ids.a % PRIME) < range_check_builtin.bound else 1 %}
        ap += 1;
        let is_not_done = [ap - 1];
```